app/_features/room/components/conference-lobby.tsx

```diff
@@ -189,13 +189,24 @@ export default function ConferenceLobby({ roomID }: LobbyProps) {
 
   const isError = !clientID || !clientName;
 
+  useEffect(() => {
+    if (isError) {
+      const timer = setTimeout(() => {
+        window.location.reload();
+      }, 5000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [isError]);
+
```

Comment:
@GagahGK 
I think this is still not a proper solution when we want to check and inform a room is not ready and still being prepared. In my previous implementation, when the `clientID` is not available, that means the room has error. But, in current case there are two possibilities when `clientID` is not available:
1. Register client endpoint error which caused the client cannot getting the client ID.
2. Room is not ready which probably caused the client ID registration error. 

We cannot inform the room is still being prepared to the user because `clientID` is not available when the **actual fact** is the room has an error because register client endpoint error. This endpoint error happens often, so please separate the logic to check `clientID` and make sure to handle register client endpoint error and when the room is not ready differently.

Reply:
When the room is not available yet, I still don't know whether the room data from hub API is already accessible to the client. Have you tested and able to simulate when the room is not available? Should we check the hub API room data to ensure the room is available or not available yet?

Reply:
The register endpoint error results in client not getting cliendID which caused by room is not yet created in the hub api or the hub it self is down

even if somehow the room have been created, and the client registration fail, the only thing to do from the client side is to refresh the page so that it will trigger : 
1. recreation of the room since this page calls `/api/room/[id]/join` which will try to check if room exist and recreate the room if not exist
2. trigger the `withRoomMiddleware` which calls the `api/room/[id]/register`

those two action will try to resolve both problem, no need to treat them differently

while for the message I think we don't have to inform the specific error to the user to let them know the technicalities

`Preparing Room` is still a correct message as it implies : 
1. the room is still creating
2. the room is registering / yet to receive register client request

Reply:
I see. If that's the case, when the `clientID` registration fails, the room will be auto reloaded every 5 seconds until the `clientID` can be generated from the hub API. I think, this is still a good approach to me.